go-controller/pkg/config/utils.go

```diff
@@ -103,6 +105,16 @@ func ParseClusterSubnetEntriesWithDefaults(clusterSubnetCmd string, ipv4HostLeng
 				return nil, NewInvalidIPv4HostSubnetError()
 			}
 
+			if !ipv6 {
+				if ipv4HostSubnet == 0 {
+					// this is the first ipv4 entry we are processing, record its host subnet length
+					ipv4HostSubnet = parsedClusterEntry.HostSubnetLength
+				} else if parsedClusterEntry.HostSubnetLength != ipv4HostSubnet {
+					return nil, fmt.Errorf("all IPv4 cluster subnet entries must have the same host subnet length; found %d and %d",
+						ipv4HostSubnet, parsedClusterEntry.HostSubnetLength)
+				}
+			}
+
```

Comment:
like you said we should look into fixing this in the future. I'm not sure why we even use src-ip routes in ovn_cluster_router. There was a proposal to modify OVN behavior to be able to prioritize src-ip over dst-ip, which I think is what we should do. But if we can avoid using src-ip routes at all that would be better. I'll follow up with you later.